animation_nodes/nodes/geometry/c_utils.pyx

```diff
@@ -0,0 +1,494 @@
+    Vector3DList, DoubleList, VirtualDoubleList,
+    VirtualVector3DList, BooleanList, CharList)
+
+cdef intersectLineLine(Vector3 *firstLineStart, Vector3 *firstLineEnd,
+                        Vector3 *secondLineStart, Vector3 *secondLineEnd,
+                        Vector3 *outFirstNearestPoint, Vector3 *outSecondNearestPoint,
+                        double *outFirstParameter, double *outSecondParameter, bint *outValid):
+    cdef Vector3 direction1, direction2, normal, normal1, normal2, direction12, direction21
+    cdef Vector3 scaledDirection1, scaledDirection2, nearestPointA, nearestPointB
+    cdef double factor1, factor2
+    subVec3(&direction1, firstLineEnd, firstLineStart)
+    subVec3(&direction2, secondLineEnd, secondLineStart)
+    crossVec3(&normal, &direction1, &direction2)
+    if lengthVec3(&normal) > 1e-6:
+        crossVec3(&normal1, &direction1, &normal)
+        crossVec3(&normal2, &direction2, &normal)
+        subVec3(&direction12, firstLineStart, secondLineStart)
+        subVec3(&direction21, secondLineStart, firstLineStart)
+        factor1 = dotVec3(&direction21, &normal2)/ dotVec3(&direction1, &normal2)
+        factor2 = dotVec3(&direction12, &normal1)/ dotVec3(&direction2, &normal1)
+        scaleVec3(&scaledDirection1, &direction1, factor1)
+        scaleVec3(&scaledDirection2, &direction2, factor2)
+        addVec3(&nearestPointA, firstLineStart, &scaledDirection1)
+        addVec3(&nearestPointB, secondLineStart, &scaledDirection2)
+        outFirstNearestPoint[0] = nearestPointA
+        outSecondNearestPoint[0] = nearestPointB
+        outFirstParameter[0] = factor1
+        outSecondParameter[0] = factor2
+        outValid[0] = True
+    else:
+        outFirstNearestPoint[0] = Vector3(0,0,0)
+        outSecondNearestPoint[0] = Vector3(0,0,0)
+        outFirstParameter[0] = 0
+        outSecondParameter[0] = 0
+        outValid[0] = False
+
+def intersectLineLineList(Py_ssize_t amount,
+                        VirtualVector3DList firstLineStartList,
+                        VirtualVector3DList firstLineEndList,
+                        VirtualVector3DList secondLineStartList,
+                        VirtualVector3DList secondLineEndList):
+    cdef Vector3DList firstNearestPointList = Vector3DList(length = amount)
+    cdef Vector3DList secondNearestPointList = Vector3DList(length = amount)
+    cdef DoubleList firstParameterList = DoubleList(length = amount)
+    cdef DoubleList secondParameterList = DoubleList(length = amount)
+    cdef BooleanList validList = BooleanList(length = amount)
+    cdef Vector3 firstNearestPoint, secondNearestPoint
+    cdef double firstParameter, secondParameter
+    cdef bint valid
+    cdef Py_ssize_t i
+    for i in range(amount):
+        intersectLineLine(firstLineStartList.get(i), firstLineEndList.get(i),
+                        secondLineStartList.get(i), secondLineEndList.get(i),
+                        &firstNearestPoint, &secondNearestPoint,
```

Comment:
Instead of `&firstNearestPoint` you could just pass `firstNearestPointList.data + i`.
Same for the other cases parameters.